src/feeds.js

```diff
@@ -308,64 +306,86 @@ feeds.fetchEventsFromCalendar_ = function(feed, callback) {
     }
     _gaq.push(['_trackEvent', 'Fetch', 'Events']);
 
-    $.ajax(feedUrl, {
-      headers: {'Authorization': 'Bearer ' + authToken},
-      success: (function(feed) {
-        return function(data) {
-          background.log('Received events, now parsing.', feed.title);
-          var events = [];
-          for (var i = 0; i < data.items.length; i++) {
-            var eventEntry = data.items[i];
-            var start = utils.fromIso8601(eventEntry.start.dateTime || eventEntry.start.date);
-            var end = utils.fromIso8601(eventEntry.end.dateTime || eventEntry.end.date);
-
-            var responseStatus = '';
-            if (eventEntry.attendees) {
-              for (var attendeeId in eventEntry.attendees) {
-                var attendee = eventEntry.attendees[attendeeId];
-                if (attendee.self) {
-                  // Of all attendees, only look at the entry for this user (self).
-                  responseStatus = attendee.responseStatus;
-                  break;
-                }
+    var fromDate = moment();
+    var fetchEventsRecursively = function(days) {
+      background.log('fetchEventsRecursively()', feed.title);
+
+      var toDate = moment().add('days', days);
+      var feedUrl =
+          feeds.CALENDAR_EVENTS_API_URL_.replace('{calendarId}', encodeURIComponent(feed.id)) + ([
+            'timeMin=' + encodeURIComponent(fromDate.toISOString()),
+            'timeMax=' + encodeURIComponent(toDate.toISOString()), 'maxResults=500',
+            'orderBy=startTime', 'singleEvents=true'
+          ].join('&'));
+
+      $.ajax(feedUrl, {
+        headers: {'Authorization': 'Bearer ' + authToken},
+        success: (function(feed) {
+          return function(data) {
+            if (data.items.length == 0) {
+              var nextInterval = days + feeds.DAYS_IN_AGENDA_;
+              if (nextInterval < feeds.MAX_DAYS_IN_AGENDA_) {
+                fetchEventsRecursively(nextInterval);
+                return;
```

Comment:
This is the main new logic, the rest of this function was just indented further.

Reply:
What you’ve written here would make an excellent inline comment!